pkg/operator/ceph/controller/spec.go

```diff
@@ -51,6 +54,32 @@ type daemonConfig struct {
 
 var logger = capnslog.NewPackageLogger("github.com/rook/rook", "ceph-spec")
 
+var (
+	cronLogRotate = `
+set -xe
+
+CEPH_CLIENT_ID=%s
+PERIODICITY=%s
+LOG_ROTATE_CEPH_FILE=/etc/logrotate.d/ceph
+
+if [ -z "$PERIODICITY" ]; then
+	PERIODICITY=24h
+fi
+
+# edit the logrotate file to only rotate a specific daemon log
+# otherwise we will logrotate log files without reloading certain daemons
+# this might happen when multiple daemons run on the same machine
+sed -i "s|*.log|$CEPH_CLIENT_ID.log|" "$LOG_ROTATE_CEPH_FILE"
+
+while true; do
+	echo "starting log rotation"
+	logrotate --verbose --force "$LOG_ROTATE_CEPH_FILE"
+	echo "I am going to sleep now, see you in $PERIODICITY"
+	sleep "$PERIODICITY"
```

Comment:
I wonder if we need to add more smarts to this. If a daemon is crash happy, does that cause the whole pod to restart, or just its container? We could see logs rotated very often for pods that crash often. An alternate could be to check the modified time of the last rotated log, rotate if it's been longer than PERIODICITY, or sleep for PERIODICITY - ( TIMENOW - last_modified ) if not.

This also might make it annoying to look at logs when daemons are restarted. I imagine there is a bit of a race condition where a daemon starts and begins logging to the end of log A, the sidecar then starts and rotates log A, and the daemon continues logging to log B. Or does `logrotate` restart the Ceph daemon?

Reply:
AFAIR, if let's say the "osd" container crashes, only it will restart, the sidecar logger we will **not** restart. I'll verify for sure, but I think I've already did. I'll confirm once and for all tomorrow.

`logrotate` does **not** restart the daemon, at least Ceph daemons respond appropriately to SIGHUP by just reloading but staying alive.
One optimization can be to wait for the daemon to be present before starting logging, I'll look into readyness checks, or else I'll do that in that bash script.

Sounds good?